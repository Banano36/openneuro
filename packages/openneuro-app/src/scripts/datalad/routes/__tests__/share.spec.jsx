import React from 'react'
import { shallow } from 'enzyme'
import { PermissionRow } from '../share.jsx'

describe('Share dataset route', () => {
  describe('PermissionRow', () => {
    it('renders for admin permissions', () => {
      expect(
        shallow(
          <PermissionRow
            datasetId="ds00001"
            userId="1234"
            userEmail="tester@example.com"
            access="admin"
          />,
        ),
      ).toMatchSnapshot()
    })
    it('renders for ro permissions', () => {
      expect(
        shallow(
          <PermissionRow
            datasetId="ds00001"
            userId="1234"
            userEmail="tester@example.com"
            access="ro"
          />,
        ),
      ).toMatchSnapshot()
    })
    it('renders for rw permissions', () => {
      expect(
        shallow(
          <PermissionRow
            datasetId="ds00001"
            userId="1234"
            userEmail="tester@example.com"
            access="rw"
          />,
        ),
      ).toMatchSnapshot()
    })
  })
})
