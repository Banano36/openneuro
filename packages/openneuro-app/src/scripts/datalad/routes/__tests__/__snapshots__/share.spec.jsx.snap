// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share dataset route PermissionRow renders for admin permissions 1`] = `
<tr>
  <td
    className="col-xs-4"
  >
    tester@example.com
  </td>
  <td
    className="col-xs-2"
  >
    Edit dataset and edit permissions
  </td>
  <td
    className="col-xs-2"
  >
    <RemovePermissions
      datasetId="ds00001"
      userId="1234"
    />
  </td>
</tr>
`;

exports[`Share dataset route PermissionRow renders for ro permissions 1`] = `
<tr>
  <td
    className="col-xs-4"
  >
    tester@example.com
  </td>
  <td
    className="col-xs-2"
  >
    View dataset
  </td>
  <td
    className="col-xs-2"
  >
    <RemovePermissions
      datasetId="ds00001"
      userId="1234"
    />
  </td>
</tr>
`;

exports[`Share dataset route PermissionRow renders for rw permissions 1`] = `
<tr>
  <td
    className="col-xs-4"
  >
    tester@example.com
  </td>
  <td
    className="col-xs-2"
  >
    Edit dataset
  </td>
  <td
    className="col-xs-2"
  >
    <RemovePermissions
      datasetId="ds00001"
      userId="1234"
    />
  </td>
</tr>
`;
